packages/amplication-client/src/VersionControl/use-buildWatchStatus.tsx

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useMemo } from "react";
+import { gql } from "apollo-boost";
+import { useQuery } from "@apollo/react-hooks";
+
+import * as models from "../models";
+
+const GENERATE_STEP_NAME = "GENERATE_APPLICATION";
+const BUILD_DOCKER_IMAGE_STEP_NAME = "BUILD_DOCKER";
+
+const POLL_INTERVAL = 1000;
+/**
+ * Pulls updates of the build from the server as long as the build process is still active
+ */
+const useBuildWatchStatus = (build: models.Build) => {
+  //check if the build process completed
+  const buildProcessCompleted = useMemo(() => {
+    if (!build.action?.steps?.length) return false;
+    const steps = build.action.steps;
+
+    const result =
+      steps.find((step) => step.name === GENERATE_STEP_NAME) &&
+      steps.find((step) => step.name === BUILD_DOCKER_IMAGE_STEP_NAME) &&
+      steps.every((step) => step.completedAt);
+    return result;
+  }, [build]);
```

Comment:
Move this logic to the backend.

Reply:
ok, will do so in the next PR since I already made changes to the logic in the next PR